raiden/network/transport/matrix/client.py

```diff
@@ -267,15 +272,25 @@ def create_sync_filter(self, broadcast_rooms: Dict[str, Room]) -> None:
             "presence": {"types": ["m.presence"]},
         }
 
+        sync_filter_id = self._create_sync_filter(filter_params)
+        return sync_filter_id
+
+    def create_timeline_limit_filter(self, limit: int) -> Optional[int]:
+        filter_params = {"room": {"timeline": {"limit": limit}}}
+        sync_filter_id = self._create_sync_filter(filter_params)
+        return sync_filter_id
```

Comment:
I'd suggest to unify the `create_broadcast_sync_filter` and `create_timeline_limit_filter` methods.
It can take both a `limit` and `broadcast_rooms` parameters.